modules/charon-core/src/main/java/org/wso2/charon/core/protocol/endpoints/UserResourceEndpoint.java

```diff
@@ -556,8 +557,105 @@ public SCIMResponse updateWithPUT(String existingId, String scimObjectString,
         }
     }
 
+    @Override
     public SCIMResponse updateWithPATCH(String existingId, String scimObjectString, String inputFormat, String outputFormat, UserManager userManager) {
-        return null;
+        //needs to validate the incoming object. eg: id can not be set by the consumer.
+
+        Encoder encoder = null;
+        Decoder decoder = null;
+
+        try {
+            //obtain the encoder matching the requested output format.
+            encoder = getEncoder(SCIMConstants.identifyFormat(outputFormat));
+            //obtain the decoder matching the submitted format.
+            decoder = getDecoder(SCIMConstants.identifyFormat(inputFormat));
+
+            SCIMResourceSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+
+            JSONObject decodedJsonObj = new JSONObject(new JSONTokener(scimObjectString));
+
+            Object metaAttributeValueObject = decodedJsonObj.opt("meta");
+            String[] metaAttributeIds = new String[0];
+
+            if (metaAttributeValueObject != null) {
+                String metaAttributeVal = decodedJsonObj.opt("meta").toString();
+                JSONObject attributeObject = new JSONObject(new JSONTokener(metaAttributeVal));
+                JSONArray attributes = (JSONArray) attributeObject.opt("attributes");
+                metaAttributeIds = new String[attributes.length()];
+                for (int i = 0; i < attributes.length(); i++) {
+                    metaAttributeIds[i] = attributes.getString(i);
+                }
+                decodedJsonObj.remove("meta");
+                scimObjectString = decodedJsonObj.toString();
+
+            }
+            //decode the SCIM User object, encoded in the submitted payload.
+            User user = (User) decoder.decodeResource(scimObjectString, schema, new User());
+            User updatedUser = null;
+            if (userManager != null) {
+                //retrieve the old object
+                User oldUser = userManager.getUser(existingId);
+                if (oldUser != null) {
+                    User validatedUser =
+                            (User) ServerSideValidator.validateUpdatedSCIMObject(oldUser, user,
+                                    schema);
+                    updatedUser = userManager.patchUser(validatedUser, oldUser, metaAttributeIds);
+
+                } else {
+                    String error = "No user exists with the given id: " + existingId;
+                    logger.error(error);
+                    throw new ResourceNotFoundException(error);
+                }
+
+            } else {
+                String error = "Provided user manager handler is null.";
+                logger.error(error);
+                throw new InternalServerException(error);
```

Comment:
Is there any specific reason for doing both logging the exception and throwing ?